device-server/src/main/kotlin/com/badoo/automation/deviceserver/host/AndroidEmulatorsNode.kt

```diff
@@ -0,0 +1,352 @@
+package com.badoo.automation.deviceserver.host
+
+import com.badoo.automation.deviceserver.data.*
+import com.badoo.automation.deviceserver.host.management.errors.DeviceCreationException
+import com.badoo.automation.deviceserver.host.management.errors.DeviceNotFoundException
+import com.badoo.automation.deviceserver.host.management.errors.OverCapacityException
+import okhttp3.*
+import org.slf4j.LoggerFactory
+import java.net.HttpURLConnection
+import java.net.URI
+import java.net.URL
+import java.util.*
+import java.util.concurrent.*
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.RuntimeException
+
+typealias EmulatorSerial = String
+
+// Represents one ticket-machine = ticketHost = machine:8088.
+class AndroidEmulatorsNode(
+        private val ticketHost: String,
+        private val emulatorLimit: Int,
+        private val concurrentBoots: Int,
+        private val concurrentBooter: ExecutorService = Executors.newFixedThreadPool(concurrentBoots),
+        private val httpClient: OkHttpClient = OkHttpClient()
```

Comment:
Let's set timeouts on HTTP client explicitly.

Reply:
Done. But WDA and CommonHttpClient have 5 and 10 seconds so I made up some numbers.